src/Symfony/Component/Cache/Traits/AbstractTrait.php

```diff
@@ -301,9 +301,9 @@ public static function handleUnserializeCallback($class)
         throw new \DomainException('Class not found: '.$class);
     }
 
-    private static function createNewNamespaceVersion(): string
+    private static function createNewNamespaceVersion(int $randomness): string
     {
-        $namespaceVersion = substr(bin2hex(pack('V', mt_rand())), 0, 8);
+        $namespaceVersion = substr(bin2hex(pack('V', $randomness)), 0, 8);
         CacheItem::validateKey($namespaceVersion);
```

Comment:
I'm still not sold about this and the need to have a method for it :)
Validating looks useless: we know that only allowed chars will be generated.